smoothie.js

```diff
@@ -671,6 +671,11 @@
       var timeSeries = this.seriesSet[d].timeSeries,
           dataSet = timeSeries.data,
           seriesOptions = this.seriesSet[d].options;
+          
+      if (seriesOptions.lineWidth === 0) {
+        context.restore();
+        continue;
```

Comment:
Indeed, that would be the effect.

Just changed it to the line 783 where it makes more sense (i was setting lineWidth to 0 and stroke was still there). That should solve the problem of having 0 width lines with fillStyles.